### ISSUE TITLE
[Phase 3] Solubility Prediction (ESOL Benchmark)

### ISSUE THREAD

**Issue by Sakeeb91** [Sakeeb91/molecular-property-prediction#4]
## Objective

Train and evaluate solubility regression model on ESOL benchmark.

## Dependencies

- Phase 2: Descriptor Generation (requires molecular features)

## Tasks

- [ ] Implement `src/models/regressor.py` with property regressor class
- [ ] Implement `src/evaluation/metrics.py` for RMSE, R2, MAE
- [ ] Use MoleculeNet scaffold split for fair evaluation
- [ ] Train models (Random Forest, XGBoost)
- [ ] Feature selection for small dataset
- [ ] Save trained model to `models/esol_model_v1.joblib`
- [ ] Write unit tests for regressor

## Files to Create

| File | Purpose |
|------|---------|
| `src/models/__init__.py` | Subpackage init |
| `src/models/regressor.py` | Property regressor |
| `src/evaluation/__init__.py` | Subpackage init |
| `src/evaluation/metrics.py` | Evaluation metrics |
| `tests/test_regressor.py` | Model tests |

## Target Metrics

| Metric | Target |
|--------|--------|
| RMSE | < 1.0 |
| R2 | > 0.7 |

## Starter Code

```python
# src/models/regressor.py
"""Molecular property regression models."""

import numpy as np
import joblib
from sklearn.ensemble import RandomForestRegressor
from xgboost import XGBRegressor

class PropertyRegressor:
    """Regressor for molecular property prediction."""

    def __init__(self, model_type: str = "xgboost", **kwargs):
        if model_type == "xgboost":
            self.model = XGBRegressor(
                n_estimators=100,
                max_depth=6,
                learning_rate=0.1,
                random_state=42,
                **kwargs
            )
        else:
            self.model = RandomForestRegressor(
                n_estimators=100,
                random_state=42,
                **kwargs
            )

    def fit(self, X: np.ndarray, y: np.ndarray) -> "PropertyRegressor":
        self.model.fit(X, y)
        return self

    def predict(self, X: np.ndarray) -> np.ndarray:
        return self.model.predict(X)

    def save(self, path: str) -> None:
        joblib.dump(self.model, path)
```

## Definition of Done

- [ ] RMSE < 1.0 on scaffold split test set
- [ ] Model beats random baseline significantly
- [ ] Feature importance analyzed
- [ ] Results comparable to MoleculeNet leaderboard
- [ ] All tests passing

## Technical Notes

- ESOL is small (~1100 molecules) - overfitting risk
- Scaffold split is harder than random split
- Feature selection crucial for small datasets

---
Part of #1 (Meta Issue)
